File: luogu/p3383.cpp
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    std::ios::sync_with_stdio(0);
    const int MAX = 100000000;
    vector<bool> lst(MAX + 1, true);
    lst[0] = lst[1] = false; // 0 和 1 不是素数

    for (int i = 2; i <= sqrt(MAX); ++i) {
        if (!lst[i]) {
            continue;
        }
        for (long long j = i; i * j <= MAX; ++j) {
            lst[i * j] = false;
        }
    }

    vector<int> primes;
    for (int i = 0; i <= MAX; ++i) {
        if (lst[i]) {
            primes.push_back(i);
        }
    }

    int n, ntc;
    cin >> n >> ntc;
    for (int i = 0; i < ntc; ++i) {
        int inp;
        cin >> inp;
        cout << primes[inp - 1] << endl; // 素数的索引是从 0 开始的
    }

    return 0;
}

File: luogu/p3383.py
import math

lst = [False, False] + [True for i in range(10**8)]
for i in range(2, math.ceil(math.sqrt(10**8))):
    if not lst[i]:
        continue
    for j in range(i, 10**8):

        if i * j > 10**8:
            break
        lst[i * j] = False

lst = list(map(lambda x: x[0], filter(lambda x: x[1], enumerate(lst))))
ntc = int(input())
for _ in range(ntc):
    inp=int(input())
    print(lst[inp-1])